Week1/homework/app.js

```diff
@@ -1,11 +1,103 @@
-'use strict';
-{
-  const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets'
-  ];
+'use strict'
+const bookKeys = [
+    "tales",
+    "oliver_twist",
+    "hamlet",
+    "monte_cristo",
+    "king_lear",
+    "mysterious_island",
+    "nora",
+    "dorian_gray ",
+    "intrigue ",
+    "search_castaways,"
+];
+
+
+
+
+'use strict'
+const bookInfo = {
+  tales: { title: "Black cat", language:"english", author: "Edgar Po"},
+  oliver_twist: { title: "Oliver Twist", language:"english", author: "Charles Dickens"},
+  hamlet: { title: "Hamlet", language:"english", author: "William Shakespeare"},
+  monte_cristo: { title: "The Count of Monte Cristo", language:"french", author: "Alexandre Dumas"},
+  king_lear: { title: "King Lear", language:"armenian", author: "William Shakespeare"},
+  mysterious_island: { title: "The Mysterious Island", language:"russian", author: "Jules Verne"},
+  nora: { title: "Nora", language:"russian", author: "Henrik Ibsen"},
+  dorian_gray: { title: "the Picture of Dorian Gray", language:"armenian", author: "Oscar Wilde"},
+  intrigue: { title: "Intrigue and Love", language:"german", author: "Friedrich Schiller"},
+  search_castaways: { title: "Search of the castaways", language:"french", author: "Jules Verne"}
+      
+          };
+          
+'use strict' 
+const covers = { tales: "https://rikrawling.files.wordpress.com/2017/06/poe-tales-of-m-and-i.jpg",
+oliver_twist: "https://i.pinimg.com/736x/17/22/98/172298024e26b1d01bee80871305254d--oliver-twist-charles-dickens.jpg",
+hamlet: "https://images-na.ssl-images-amazon.com/images/I/51VQaOw9dKL._SX331_BO1,204,203,200_.jpg",
+monte_cristo:"https://images-na.ssl-images-amazon.com/images/I/51W5xvphHlL.jpg",
+king_lear: "https://images-nitrosell-com.akamaized.net/product_images/8/1889/large-king%20lear%20forum%20new%20ed.jpg",
+mysterious_island: "https://images-na.ssl-images-amazon.com/images/I/81ySxuhHRQL.__BG0,0,0,0_FMpng_AC_UL320_SR200,320_.jpg",
+nora: "https://images-na.ssl-images-amazon.com/images/I/41FZWVMABKL._SX318_BO1,204,203,200_.jpg",
+dorian_gray: "https://a.wattpad.com/cover/6626233-352-k168522.jpg",
+intrigue: "https://images-na.ssl-images-amazon.com/images/I/51N2vwu3tdL.jpg",
+search_castaways: "http://prodimage.images-bn.com/pimages/9781534866072_p0_v1_s1200x630.jpg"
+};
+
+'use strict'
+  let list = document.createElement("LI");
+ 
+function showBook() {
+
+for (let [k, z] of Object.entries(bookInfo)){
+
+  
+  
+
+    let header_2 = document.createElement("H2");
+    let textnode = document.createTextNode(k + " - " +  z.title + "  " + z.language  + "  " + z.author);
```

Comment:
I agree with you `Object.entries` is a very useful feature in Javascript, But what if you get this situation that `bookInfo` returned from a server an empty object, it doesn't have anything?
in this case you have `[k, z]` and both of them are undefined and in this line exactly where you will got an error back that says __`TypeError: z.title is undefined`__ and if you are expecting a function inside of this your whole app will just stop.

Also no problem using `Object.entries` But as a person of people for the first time I will just look to the code and be confused of the new pattern I see, and `Object.keys` I am not talking about taste :upside_down_face: but I am talking about legibility *(sometimes it looks better)*, so following the mostly used pattern is very useful on the low level as a homework and on the high level as a worldwide of developers.

so instead you can use `Object.keys` as the following: 
```javascript
function myBeautifulFunction() {
  const bookInfoKeys = Object.keys(bookInfo);
  // for avoiding errors
  if (!bookInfoKeys.length) return;
  for(let key of bookInfoKeys){
    const title = key.title;
    const author = key.author;
    // etc...
  }
}
```

`Object.entries` way: 
```javascript
function myBeautifulFunction() {
  const bookInfoKeys = Object.entries(bookInfo);
  // for avoiding errors
  if (!bookInfoKeys.length) return;
  for(let [k, z] of bookInfoKeys){
    // keys are there but the values are empty we have to check again
    if (!z.title) return;
    // what about human readable names?
    // Your code...
  }
}
```